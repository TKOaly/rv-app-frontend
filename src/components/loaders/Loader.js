import './styles/Loader.css';
import React from 'react';

const Loader = () => <div className="loader" />;

export default Loader;
