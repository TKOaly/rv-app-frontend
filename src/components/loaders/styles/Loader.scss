@import "vars";

.loader {
  border: 2px solid transparent;
  border-top: $loader-width solid $loader-color;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  animation: spin 0.5s linear infinite;
  margin-left: auto;
  margin-right: auto;
}
@keyframes spin {
  
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
