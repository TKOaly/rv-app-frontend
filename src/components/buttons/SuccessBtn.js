import React from 'react';
import './styles/Button.css';
import './styles/SuccessBtn.css';
import Loader from './../loaders/Loader';
import PropTypes from 'prop-types';

const SuccessBtn = ({ onClick, children, fill, hover, loader, ...props }) => {
    let className = 'btn';
    fill ? (className += ' success-fill') : (className += ' success');
    hover && (fill ? (className += ' success-fill-hover') : (className += ' success-hover'));

    return (
        <button {...props} onClick={onClick} className={className}>
            {!loader ? <span className="btnContent">{children}</span> : <Loader />}
        </button>
    );
};

SuccessBtn.propTypes = {
    /** Click handler. */
    onClick: PropTypes.func,
    /** Button fill. If set to `true`, the button's colors will "invert". */
    fill: PropTypes.bool,
    /** Hover effect. If set to `true`, the button will have a hover effect. */
    hover: PropTypes.bool,
    /** Loader effect. If set to `true`, will show a CSS loader instead of button text. */
    loader: PropTypes.bool
};

SuccessBtn.defaultProps = {
    onClick: null,
    fill: false,
    hover: false,
    loader: false
};

export default SuccessBtn;
