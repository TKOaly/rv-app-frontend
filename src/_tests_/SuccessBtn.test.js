import { shallow } from 'enzyme';
import Loader from '../components/loaders/Loader';
import React from 'react';
import SuccessBtn from '../components/buttons/SuccessBtn';
import renderer from 'react-test-renderer';

describe.only('<SuccessBtn />', () => {
    it('renders text correctly', () => {
        const data = {
            text: 'Hello world'
        };

        const successBtnComponent = shallow(<SuccessBtn>{data.text}</SuccessBtn>);
        console.log(successBtnComponent.debug());
        const contentDiv = successBtnComponent.find('.btnContent');
        expect(contentDiv.text()).toContain(data.text);
        const cmpnt = renderer.create(<SuccessBtn>{data.text}</SuccessBtn>).toJSON();
        expect(cmpnt).toMatchSnapshot();
    });

    it('shows loader correctly', () => {
        const successBtnComponent = shallow(<SuccessBtn loader={true} />);
        console.log(successBtnComponent.debug());
        expect(successBtnComponent.contains(<Loader />)).toBe(true);
        const cmpnt = renderer.create(<SuccessBtn loader />).toJSON();
        expect(cmpnt).toMatchSnapshot();
    });

    it('onClick event works correctly', () => {
        const data = {
            text: 'Hello world'
        };

        const mockHandler = jest.fn();

        const successBtnComponent = shallow(<SuccessBtn onClick={mockHandler} />);

        const cmpnt = renderer.create(<SuccessBtn onClick={mockHandler} />).toJSON();
        expect(cmpnt).toMatchSnapshot();

        const button = successBtnComponent.find('button');

        expect(mockHandler.mock.calls.length).toBe(0);
        button.simulate('click');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
});
