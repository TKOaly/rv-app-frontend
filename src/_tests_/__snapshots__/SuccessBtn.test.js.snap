// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessBtn /> onClick event works correctly 1`] = `
<button
  className="btn success"
  onClick={[MockFunction]}
>
  <span
    className="btnContent"
  />
</button>
`;

exports[`<SuccessBtn /> renders text correctly 1`] = `
<button
  className="btn success"
  onClick={null}
>
  <span
    className="btnContent"
  >
    Hello world
  </span>
</button>
`;

exports[`<SuccessBtn /> shows loader correctly 1`] = `
<button
  className="btn success"
  onClick={null}
>
  <div
    className="loader"
  />
</button>
`;
